[Mind-22] 매칭 요청 취소 기능 구현

TaskerJang opened the pull request:
# Pull Request

## 💡 PR 요약
매칭 취소 기능을 구현했습니다. 사용자가 이전에 요청한 매칭을 취소할 수 있는 API를 개발했습니다.

## 🔍 주요 변경사항
- MatchingService에 매칭 취소 기능 추가
- MatchingController에 매칭 취소 API 엔드포인트(GET /api/v1/matching/{id}) 구현
- BaseResponseStatus에 매칭 취소 관련 상태 코드 추가
- 매칭 취소 기능에 대한 단위 테스트 코드 작성

## 🔗 연관된 이슈


## ✅ 체크리스트
- [x] 테스트 코드를 작성하였나요?
- [x] 관련 문서를 업데이트하였나요?
- [ ] Breaking Change가 있나요?
- [x] 코드 포맷팅과 린트 검사를 완료하였나요?

## 🙏 리뷰어 참고사항


## 📋 추가 컨텍스트

Commit:
[feat]: 매칭 취소 기능 구현
- 사용자가 요청한 매칭을 취소하는 API 구현
- 매칭 취소 시 상태 CANCELED로 변경
- 취소 권한 검증 및 에러 처리

Commit:
[test]: 매칭 취소 기능 테스트 코드 추가
- 매칭 취소 성공 케이스 테스트
- 존재하지 않는 매칭 취소 실패 케이스 테스트
- 권한 없는 사용자의 취소 실패 케이스 테스트
- 이미 취소된 매칭 취소 실패 케이스 테스트

hwan2-99 commented on src/main/java/com/mindgoal/domain/matching/controller/MatchingController.java:
```diff
@@ -1 +1,55 @@
+     * @param principalDetails 인증된 사용자 정보
+     * @return 취소된 매칭 정보
+     */
+    @GetMapping("/{id}")
```
id라고 하니 처음에 어떤 id값인지 이해하기 어려웠던것 같습니다! 매칭의 id값을 나타낸거면 matchingId 같이 명확하게 나타내면 어떨까요??

hwan2-99 commented on src/main/java/com/mindgoal/domain/matching/service/MatchingService.java:
```diff
@@ -1 +1,69 @@
+    @Transactional
+    public MatchingResponseDto cancelMatching(Long matchingId, Long userId) {
+        // 매칭 존재 여부 확인
+        Matching matching = matchingRepository.findById(matchingId)
```
해당부분도 메서드로 분리하면 좋을것 같습니다!

hwan2-99 commented on src/main/java/com/mindgoal/domain/matching/service/MatchingService.java:
```diff
@@ -1 +1,69 @@
+        // 이미 취소된 매칭인지 확인
+        if (matching.getStatus().equals(MatchingStatus.CANCELED.name())) {
+            throw new CustomException(BaseResponseStatus.MATCHING_ALREADY_CANCELED);
+        }
```
아마 점점 예외처리 코드가 늘어날것 같습니다! 아래처럼 변경해보는건 어떨까요?? 그럼 예외처리가 늘어날수록 해당메서드가 두꺼워지지않고 단일책임원칙이 지켜질것 같습니다!
```java
@Transactional
public MatchingResponseDto cancelMatching(Long matchingId, Long userId) {
  Matching matching = matchingRepository.findById(matchingId) .orElseThrow(() -> new CustomException(BaseResponseStatus.MATCHING_NOT_FOUND));
validateMatching(matching,userId);
....
}

private void validateMatching(Matching matching, Long userId) {
    validateMatchingOwner(matching, userId);
    validateMatchingStatus(matching);
}

private void validateMatchingOwner(Matching matching, Long userId) {
    if (!matching.getUserId().equals(userId)) {
        throw new CustomException(BaseResponseStatus.UNAUTHORIZED_ACCESS);
    }
}

private void validateMatchingStatus(Matching matching) {
    if (matching.getStatus().equals(MatchingStatus.CANCELED.name())) {
        throw new CustomException(BaseResponseStatus.MATCHING_ALREADY_CANCELED);
    }
}

hwan2-99 reviewed (commented):
고생하셨습니다!

Commit:
[refactor]: 매칭 관련 클래스 구조 개선

- MatchingController의 매개변수명 id → matchingId로 변경하여 명확성 향상
- MatchingService의 검증 로직 메서드 분리로 단일 책임 원칙 적용
- 계층적 검증 구조 도입으로 확장성 개선

Commit:
conflict resolve
